watson_developer_cloud/speech_to_text_v1.py

```diff
@@ -18,10 +18,133 @@
 
 from .watson_developer_cloud_service import WatsonDeveloperCloudService
 import json
+from collections import deque, Counter, namedtuple
+import base64
+
+from autobahn.twisted.websocket import WebSocketClientFactory,\
+    WebSocketClientProtocol, connectWS
+from twisted.internet import ssl, reactor, task, interfaces
+from zope.interface import implementer
+
+Filedata = namedtuple('FileData', ['file_name',
+                                   'file_metadata',
+                                   'initial_data'])
+
+
+@implementer(interfaces.IPushProducer)
+class AudioFileProducer:
+    def __init__(self, proto):
+        self.proto = proto
+        self.started = False
+        self.paused = False
+        self.filedata = proto.filedatasource.pop()
+        self.fileobj = open(self.filedata.file_name, 'rb')
+        self.proto.metadata = self.filedata.file_metadata
+
+    def pauseProducing(self):
+        self.paused = True
+
+    def resumeProducing(self):
+        self.paused = False
+        if not self.started:
+            as_json = json.dumps(self.filedata.initial_data).encode('utf-8')
+            self.proto.sendMessage(as_json, False)
+            self.started = True
+
+        while not self.paused:
+            buf = self.fileobj.read(1024)
+            if not buf:
+                as_json = json.dumps({"action": "stop"}).encode('utf-8')
+                self.proto.sendMessage(as_json, False)
+                break
+            self.proto.sendMessage(buf, True)
+
+    def stopProducing(self):
+        self.fileobj.close()
+
+
+class StreamingSpeechToTextProtocol(WebSocketClientProtocol):
+    filedatasource = deque()
+    params = {}
+    factory = None
+    ssl_context_factory = None
+    connects = 0
+
+    @classmethod
+    def buildFactory(cls, url, headers, params):
+        # only build the factory once
+        if cls.factory is None:
+            cls.factory = WebSocketClientFactory(url, headers=headers)
+            cls.factory.protocol = cls
+            cls.ssl_context_factory = ssl.ClientContextFactory()
+            cls.params = params
+
+    @classmethod
+    def start(cls):
+        reactor.run()
+
+    @classmethod
+    def stop(cls):
+        reactor.stop()
+
+    @classmethod
+    def addContentType(cls, content_type):
```

Comment:
Same as above:  ok, this code isn't part of someone elses api, however, my argument is that since it has other camelCase methods that the methods of this class should all be consistent.